Use default HttpExchange#setStreams implementation

strogiyotec opened the pull request:
Right now the `JettyHttpExchangeDelegate#setStreams` implementation just sets given parameters to respective steams. However, the default JDK implementation of `HttpExchange#setStreams` calls `ExchangeImpl#setStreams` which in turn only changes the respective streams when given parameters aren't null. To follow this convention the  `JettyHttpExchangeDelegate#setStreams` was rewritten to have the same implementation

joakime commented:
There is no valid Eclipse ECA on file for the email addresses on these commits, and as such this PR is not legally allowed to be merged.

Please review https://github.com/eclipse/jetty.project/blob/jetty-10.0.x/CONTRIBUTING.md#eclipse-contributor-agreement

strogiyotec commented:
@joakime  sorry I wasn't familiar with it , I signed the agreement , feel free to take a look at the PR when you have time

sbordet commented on jetty-http-spi/src/main/java/org/eclipse/jetty/http/spi/JettyHttpExchangeDelegate.java:
```diff
@@ -209,8 +209,15 @@ public void setAttribute(String name, Object value)
-        _is = i;
-        _os = o;
+        assert _is != null;
+        if (i != null)
+        {
+            _is = i;
+        }
+        if (o != null)
+        {
+            _os = o;
+        }
```
Drop the braces in case of one-liners.

```suggestion
        if (i != null)
            _is = i;
        if (o != null)
            _os = o;
```

strogiyotec committed:
Use default HttpExchange#setStreams implementation

Right now the JettyHttpExchangeDelegate#setStremas implementation just sets given parameters to respective steams. However, the default JDK implementation of HttpExchange#setStreams calls #ExchangeImpl.setStreams which in turn
only changes the respective streams when given parameters aren't null.
To follow this convention the  JettyHttpExchangeDelegate#setStremas was
rewritten to follow the same convention

strogiyotec commented:
@sbordet  I made the changes please take a look

sbordet commented:
@strogiyotec for the next PRs, please do not force-push, because for a reviewer it becomes harder to review, particularly with many files to review.

Thanks!

strogiyotec commented:
got it thanks for taking time and helping me

sbordet commented:
@strogiyotec Thanks!
